docs/modules/extend/examples/pdf-converter-article-title-with-author-and-date.rb

```diff
@@ -0,0 +1,43 @@
+class PDFConverterArticleTitleWithAuthorAndDate < (Asciidoctor::Converter.for 'pdf')
+  register_for 'pdf'
+
+  def ink_general_heading doc, title, opts
+    return super unless opts[:role] == :doctitle # <1>
+    ink_document_title title, opts # <2>
+    ink_document_metainfo doc
+    ink_titlesection_body_separator
+  end
+
+  def ink_document_title title, opts
+    if (top_margin = @theme.heading_h1_margin_page_top || @theme.heading_margin_page_top) > 0
+      move_down top_margin
+    end
+    pad_box @theme.heading_h1_padding do
+      if (transform = resolve_text_transform opts)
+        title = transform_text title, transform
+      end
+      if (inherited = apply_text_decoration font_styles, :heading, 1).empty?
+        inline_format_opts = true
+      else
+        inline_format_opts = [{ inherited: inherited }]
+      end
+      typeset_text_opts = { color: @font_color, inline_format: inline_format_opts }.merge opts
+      typeset_text title, (calc_line_metrics (opts.delete :line_height) || @base_line_height), typeset_text_opts 
+    end
+  end
+
+  def ink_document_metainfo doc
+    revremark = doc.attr 'revremark' # <3>
+    if doc.author || doc.revdate || revremark # <4>
+      theme_font :base do
+        author_date_separator = (doc.author && doc.revdate) ? ' – ' : '' # <5>
```

Comment:
https://github.com/asciidoctor/asciidoctor-pdf/pull/2439#issuecomment-1748391437 uses `' &#x2013; '` rather than `' – '`. Are non-ASCII characters to be avoided in the source code? I kinda like to see what text snippet in a string literal will kinda "look" like without having to memorize code point numbers, but I can understand that it might be also important to be able to spot non-ASCII characters easily and to be able to distinguish them at sight.

Reply:
> I kinda like to see what text snippet in a string literal will kinda "look" like without having to memorize code point numbers

I prefer not to rely on extended Unicode characters in the source code because it makes it unclear what the character actually is (we can all assume we can recognize ASCII characters).

If you want to make it more clear, you can extract it as a constant as I have done in the converter itself. See https://github.com/asciidoctor/asciidoctor-pdf/blob/main/lib/asciidoctor/pdf/converter.rb#L87 In that case, the preferred syntax is `?\uNNNN`.

Reply:
EmDash is actually already defined there:
https://github.com/asciidoctor/asciidoctor-pdf/blob/5f44ad256d1e7e7ca0ebcb6e1cbf1ee59b5949f8/lib/asciidoctor/pdf/converter.rb#L86-L86
So with 791f0be83d8560ebe8e3b5ff735db0991c6389ca I just use that.